modules/core/localizer.js

```diff
@@ -88,6 +89,8 @@ export function coreLocalizer() {
             'locales'     // load the list of supported locales
         ];
 
+        const presetsVersion = packageJSON.devDependencies['@openstreetmap/id-tagging-schema'];
+        const presetsCdnUrl = presetsCdnUrlTemplate.replace('{presets_version}', presetsVersion);
```

Comment:
actually… this was working as intended:

[`fileFetcher.get`](https://github.com/openstreetmap/iD/blob/1d64839b24d9c7f6d79b3810961d1f0a27c59d27/modules/core/file_fetcher.js#L67-L71) expands the "template" with the correct version of the loaded presets. Just using the version from `packageJson` doesn't always work because of caching: say you have `v5.0.1` of one of the presets' files cached, but the rest is loaded fresh from the cdn (say, `v5.2.1`), then these files might very likely mismatch in at least a few places, which can result in missing translations for example.

To avoid this, the fileFetcher does first load the preset's `package.json` and use the exact version it gets from that request to load all further preset files. This assures that all files are consistent with each other.

--

The url template can be passed to the fileFetcher:


```javascript

        const localeDirs = {
            general: 'locales',
            tagging: presetsCdnUrlTemplate + 'dist/translations'
        };
```

Reply:
@tyrasd even thought this PR is closed, I think your feedback and the fact that I was not able to read those details from the variable names hints that the variable names could be improved. The name itself you communicate if its a "template" – as in, something of the string will need to be replaced in order to work – or a URL that will be used as is.

This might be something to refactor at some point…